core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java

```diff
@@ -123,7 +123,7 @@ enum SchemaVersion {
           + JdbcTableOperations.METADATA_LOCATION_PROP
           + " = ?";
   static final String V0_CREATE_CATALOG_SQL =
-      "CREATE TABLE "
+      "CREATE TABLE IF NOT EXISTS "
```

Comment:
The reason why we use JDBC table check (via catalog) and not the `IF NOT EXISTS` statement is to work with any RDBMS.

To my knowledge, some database doesn't support `IF NOT EXISTS` statement.
`IF NOT EXISTS` has been introduced in Oracle Database 23c, and it's supported by MySQL, MariaDB, PostgreSQL and SQLite, but not by SQLServer.

So, no problem to use this approach as soon as we clearly state that SQLServer won't be supported.

Reply:
Thanks for clarifying this,  @jbonofre , How could we know whether SQLServer will be supported or not?

Reply:
We can do a quick test using SQLServer docker image (it's the way I'm testing JDBC Catalog with different RDBMS).

Reply:
I tested SQLServer and it doesn't support `CREATE TABLE IF NOT EXISTS`, seems this is not the correct direction if  Iceberg still supports SQLServer as JDBC catalog backend.  How about double check the database and table information from the `ResultSet` returned by `dbMeta.getTables`,  is it the right direction?

Reply:
Yes, it's what I said: SQLServer doesn't support it. 
I think we have two options:
1. We clearly state that SQLServer is not supported
2. I prefer to keep the current approach using `dbMeta`. I would probably add an extra check to define if tables have to be created or not.

I'm more in favour of 2 as it would work with any RDBMS.

Reply:
got it

Reply:
Another way to check if tables are there, can be done by running:

```sql
SELECT 1 FROM table
```

When it raises an exception, you must create the tables (keep in mind race conditions). This does add another query per table.

Reply:
@Fokko thanks for your advice, I like the idea,  @jbonofre WDYT?